### ISSUE TITLE
Task: Set up new `main.py` to run on the VM

### ISSUE THREAD

**Issue by nlebovits** [CodeForPhilly/clean-and-green-philly#1045]
## Describe the task
We need to get the new pipeline running on the VM. In order to do this, you'll need to update the `.cagp_env` file appropriately, build the new docker container, and update the CRON job to run `main.py` once a month. We'll also want to create monthly backups (or maybe quarterly backups?) of the database. We'll need to make sure all of our slack messages are reporting correctly for monitoring purposes, and that the data are being written to the right place in GCP to be usable by the website. Since we have the data in hypertables now, I don't think we need to be as concerned about backup up the `pmtiles,` but we should still due our due dilligence to make sure we're not deleting things. I've backed up the old postgres database to GCP inccase something happens. 

**Note: I know this ticket needs to be improved. Please throw a comment on here if you're interested and I'll come back with more context.**

## Acceptance Criteria

- [ ] first
- [ ] second
- [ ] third

## Additional context

Here's a shell script you modify as the basis of your `pg_dump` backups:

```
#!/bin/bash

# Set variables
TIMESTAMP=$(date +"%Y%m%d_%H%M%S") # Current timestamp
DB_NAME="vacantlotdb"             # Replace with your database name
DB_USER="postgres"                # Replace with your database username
DOCKER_CONTAINER="cagp-postgres"  # Replace with your PostgreSQL Docker container name
BACKUP_FILE="vacantlotdb_backup_$TIMESTAMP.sql.gz"
BUCKET="gs://cleanandgreenphl/db_backups"

# Dump the database using Docker and compress the output
echo "Starting database backup..."
docker exec $DOCKER_CONTAINER pg_dump -U $DB_USER -d $DB_NAME | gzip > /tmp/$BACKUP_FILE

# Upload the backup to GCS
echo "Uploading backup to Google Cloud Storage..."
gsutil cp /tmp/$BACKUP_FILE $BUCKET

# Verify upload success
if [ $? -eq 0 ]; then
  echo "Backup uploaded successfully."
  # Remove local backup file to save space
  rm /tmp/$BACKUP_FILE
else
  echo "Failed to upload backup to GCS."
  exit 1
fi

echo "Monthly database backup completed successfully."
```

**Comment by adamzev** [comment]
Jerry and I were discussing this at the meetup. We are both newly on-boarding to the project so we'll have to dig more into the details.

Here's the outline for what I was proposing (described in terms of AWS resources but Jerry is planning on writing the IaC in Pulumi so it should be updatable to GCP if we stick with that):

- GitHub Actions: A monthly scheduled GitHub Action starts the pipeline in AWS
- ECR: Elastic Container Registry stores the container image
- ECS Fargate: Runs the Python ETL script (same Docker commands as running it locally) using low cost compute
- the function would return success or failure triggering the retries
- S3: Store the results in a bucket

Because we want to leave room for the pipeline to expand, I think we're too close to the runtime limit to use Lambda functions instead.

**Comment by nlebovits** [comment]
This sounds great! Feel free to ping me with questions about implementation--I'll do my best to answer. Preference is definitely to stick with GCP--we'd like to keep things as consolidated as possible.

@SofiaFasullo

**Comment by nlebovits** [comment]
Also, apologies for the terrible documentation on this PR. I was in the middle of a major move at the time; happy to try to explain things in more detail when possible.

**Comment by adamzev** [comment]
No worries, thanks for the quick response. I'm trusting LLMs to rewrite this a bit more than I like here, but it looks like an equivalent workflow in GCP would be:

- GitHub Actions: A monthly scheduled GitHub Action starts the pipeline in Google Cloud.
- Artifact Registry: Stores the container image.
- Cloud Run (or Cloud Run Jobs): Runs the Python ETL script (using the same Docker commands as running it locally) with low-cost, serverless compute. The function returns success or failure, triggering retries if needed.
- Cloud Storage: Stores the results in a bucket.

**Comment by nlebovits** [comment]
Yep, this generally sounds good to me. To date, we've just been using a CRON job on a GCP VM. We have a Slackbot and email message set up for failure reporting, but the script isn't running right now anyway, so it's all on hold. It will be good to get a more robust version of the pipeline set up with the new ETL script. Lmk if/when you have questions.

**Comment by nlebovits** [comment]
While you're working on it, @adamzev can you decide if #548 is obsolete? You can either add it to your work here or, if it's no longer necessary, we can close it.

**Comment by nlebovits** [comment]
Also take a look at #984 -- I'm guessing that should also be a consideration here

**Comment by nlebovits** [comment]
Obsolete due to newer ticket.
